src/distilabel/steps/generators/huggingface.py

```diff
@@ -136,7 +154,7 @@ def process(self, offset: int = 0) -> "GeneratorStepOutput":
             transformed_batch = self._transform_batch(batch)
             batch_size = len(transformed_batch)
             num_returned_rows += batch_size
-            yield transformed_batch, num_returned_rows == num_examples
+            yield transformed_batch, num_returned_rows == self.num_examples
```

Comment:
Maybe I'm missing something, but shouldn't this be greater or equal instead?

```suggestion
            yield transformed_batch, num_returned_rows >= self.num_examples
```